File: epoxy-processortest/src/test/resources/AbstractModelWithHolder.java
package com.airbnb.epoxy;

import android.view.View;
import androidx.annotation.NonNull;

@EpoxyModelClass
public abstract class AbstractModelWithHolder
    extends EpoxyModelWithHolder<AbstractModelWithHolder.Holder> {
  @EpoxyAttribute int value;

  @Override
  protected int getDefaultLayout() {
    return 0;
  }

  static class Holder extends EpoxyHolder {

    protected void bindView(@NonNull View itemView) {

    }
  }
}

File: epoxy-processortest/src/test/resources/testModelWithHolderGeneratesNewHolderMethod/AbstractModelWithHolder.kt
package com.airbnb.epoxy

import android.view.View

@EpoxyModelClass
abstract class AbstractModelWithHolder : EpoxyModelWithHolder<Holder>() {
    @EpoxyAttribute
    var value = 0

    override fun getDefaultLayout(): Int {
        return 0
    }
}

class Holder : EpoxyHolder() {
    override fun bindView(itemView: View) {}
}
